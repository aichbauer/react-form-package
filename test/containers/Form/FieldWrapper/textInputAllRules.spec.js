import React from 'react';
import renderer from 'react-test-renderer';
import Adapter from 'enzyme-adapter-react-16';
import { configure, mount } from 'enzyme';

import {
  Form,
  FieldWrapper,
  Button,
} from '../../../../src';
import { textInputAllRules } from '../../../fixtures';

configure({ adapter: new Adapter() });

test('Render Form with text input | FieldWrapper | text input all rules', () => {
  const mock = jest.fn();
  const fieldWrapperOnChangeSpy = jest.spyOn(Form.prototype, 'handleOnChange');
  const fieldWrapperOnFocusSpy = jest.spyOn(Form.prototype, 'handleOnFocus');
  const fieldWrapperOnBlurSpy = jest.spyOn(Form.prototype, 'handleOnBlur');

  const event = {
    target: {
      id: 'text',
      value: 'hello',
    },
  };
  const rules = {
    type: 'text',
    min: '3',
    max: '8',
    required: false,
  };

  const CustomInput = (props) => (
    <input
      onChange={(e) => props.onChange(e.target.value)}
      onFocus={(e) => props.onFocus(e.target.value)}
      onBlur={(e) => props.onBlur(e.target.value)}
      meta={props.meta}
    />
  );

  const myComponent = (
    <Form>
      <div>
        <div>text</div>
        <FieldWrapper onChange={() => mock()} type="text" id="text" min="3" max="8" required>
          <CustomInput />
        </FieldWrapper>
      </div>
    </Form>
  );

  const tree = renderer.create(myComponent).toJSON();
  const wrapper = mount(myComponent);
  const exposedPropsOnCustomComponent = wrapper.find('input').props();
  let state = wrapper.state();

  expect(exposedPropsOnCustomComponent).toHaveProperty('onChange');
  expect(exposedPropsOnCustomComponent).toHaveProperty('onFocus');
  expect(exposedPropsOnCustomComponent).toHaveProperty('onBlur');
  expect(exposedPropsOnCustomComponent).toHaveProperty('meta');
  expect(state.data.text.value).toBe('');
  expect(state.data.text.valid).toBe(false);

  wrapper.find('input').simulate('focus');
  wrapper.find('input').simulate('change', event, rules);
  wrapper.find('input').simulate('blur');
  wrapper.update();

  state = wrapper.state();

  expect(state.data.text.value).toBe('hello');
  expect(state.data.text.valid).toBe(true);
  expect(fieldWrapperOnChangeSpy).toHaveBeenCalled();
  expect(fieldWrapperOnFocusSpy).toHaveBeenCalled();
  expect(fieldWrapperOnBlurSpy).toHaveBeenCalled();
  expect(mock).toHaveBeenCalled();
  expect(tree).toMatchSnapshot();
});
