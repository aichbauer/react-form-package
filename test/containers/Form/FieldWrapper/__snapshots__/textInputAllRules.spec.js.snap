// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Render Form with text input | FieldWrapper | text input all rules 1`] = `
<form>
  <div>
    <div>
      text
    </div>
    <input
      meta={
        Object {
          "checked": undefined,
          "dirty": false,
          "initialValue": "",
          "invalid": true,
          "pristine": true,
          "rules": Object {
            "match": undefined,
            "max": 8,
            "min": 3,
            "required": true,
            "sameAs": undefined,
            "type": "text",
          },
          "touched": false,
          "valid": false,
          "value": "",
          "visited": false,
        }
      }
      onBlur={[Function]}
      onChange={[Function]}
      onFocus={[Function]}
    />
  </div>
</form>
`;
